src/language_concepts/standard_library.md

```diff
@@ -0,0 +1,70 @@
+# Noir Standard Library
+
+Noir features a standard library with some ready-to-use, built-in structures. To use them, you should import the `std` library, like so:
+
+``` rust
+
+use dep::std;
+```
+
+You should then have these constructs available. Example:
+
+``` rust
+let data : [Field; 2] = [42, 42];
+std::hash::pedersen(data);
+```
+
+## Field
+
+After declaring a Field, you can use these common methods on it [^migrationNote]:
+
+| Method   | Description  | Example                        |
+|:---------|--------------|--------------------------------|
+|to_le_bits| Transforms into an array of bits, Little Endian | `field.to_le_bits()` |
+|to_le_bytes| Transforms into an array of bytes, Little Endian | `field.to_le_bytes(byte_size)` |
+|to_radix| Decomposes into a vector over the specificed base | `field.to_radix(radix_base, vector)`
+|pow_32| Returns the value to the power of the specificied exponent | `field.to_pow(2)` |
+
+## Array
+
+For convenience, the STD provides some ready-to-use, common methods for arrays[^migrationNote]:
+
+| Method   | Description    | Example                        |
+|:---------|----------------------|--------------------------------|
+| len      | Returns the length of an array                                                                                                | `array.len()`                  |
+| sort | Sorts the array with a built-in function                                                                                           | `array.sort()`                 |
+| sort_via | Sorts with a custom sorting function                                                                                          | `array.sort_via(|a, b| a < b)` |
+| fold     | Applies a function to each element of the array, returning the final accumulated value. First parameter is the initial value. | `array.fold(0, |a, b| a < b)`  |
+| reduce   | Same as fold, but uses the first element as starting element                                                                  | `array.reduce(|a, b| a < b)`   |
```

Comment:
These examples are invalid. The lambda function takes two integers and returns a Bool, but `reduce` requires the lambda's type to be `fn(T, T) -> T` - or in other words - the argument types must match the return type.

`fold` is similarly invalid. It has the same lambda type, but fold requires the lambda to be of type `fn(U, T) -> U`. This is slightly more flexible than reduce, but still invalid for the chosen lambda.

We can change these examples to `array.fold(0, |a, b| a + b)` and `array.reduce(|a, b| a + b)` to fix this.

This also highlights that we should also document the type signatures for all library functions as well, so that users know how to use them.